src/commands/android/utils/common.ts

```diff
@@ -152,3 +153,32 @@ export const downloadFirefoxAndroid = async (version: string) => {
 
   return await downloadWithProgressBar(apkDownloadUrl, tempdir);
 };
+
+export const showSubcommandHelp = (unknownSubcommand?: string) => {
+  if (unknownSubcommand) {
+    Logger.log(`${colors.red('unknown subcommand passed:')} ${unknownSubcommand}\n`);
+  }
+  Logger.log(`Usage: ${colors.cyan('npx @nightwatch/mobile-helper android [subcmd] [subcmd-options]')}`);
+  Logger.log('  The following subcommands are used for different operations on Android SDK:\n');
+  Logger.log(`${colors.yellow('Subcommands and Subcommand-Options:')}`);
+
+  const longest = (xs: string[]) => Math.max.apply(null, xs.map(x => x.length));
+
+  Object.keys(AVAILABLE_SUBCOMMANDS).forEach(subcommand => {
+    const subcmd = AVAILABLE_SUBCOMMANDS[subcommand];
+    const subcmdOptions = subcmd.options?.map(option => `[--${option.name}]`).join(' ') || '';
+
+    Logger.log(`  ${colors.cyan(subcommand)} ${subcmdOptions}`);
+    Logger.log(`  ${colors.gray(subcmd.description)}`);
+
+    if (subcmd.options && subcmd.options.length > 0) {
+      const optionLongest = longest(subcmd.options.map(option => `--${option.name}`));
+      subcmd.options.forEach(option => {
+        const optionStr = `--${option.name}`;
+        const optionPadding = new Array(Math.max(optionLongest - optionStr.length + 3, 0)).join('.');
+        Logger.log(`    ${optionStr} ${colors.grey(optionPadding)} ${colors.gray(option.description)}`);
+      });
+    }
+    Logger.log();
```

Comment:
One good thing to do here would be, instead of logging everything directly, we can just create a large string containing all the help data and then return that string.

This would allow us to prepend/append something to the string more easily as the use case might arise and even modify parts of the content by using some string manipulation methods.

Also, the function name can be changed to `getSubcommandHelp` to reflect the new behaviour.

Reply:
This change should only require replacing the `Logger.log` calls with some string append calls. Also, we can remove the `unknownSubcommand` parameter from this function and that could be handled separately in the code calling this function.